src/queries/compare.ts

```diff
@@ -0,0 +1,53 @@
+import { GitHub } from '@actions/github/lib/utils'
+import { CompareBaseHeadQuery, CompareBaseHeadQueryVariables } from '../generated/graphql'
+import { retryHttpError } from './retry'
+
+type Octokit = InstanceType<typeof GitHub>
+
+const query = /* GraphQL */ `
+  query compareBaseHead($owner: String!, $name: String!, $base: String!, $head: String!, $size: Int!, $after: String) {
+    rateLimit {
+      cost
+    }
+    repository(owner: $owner, name: $name) {
+      ref(qualifiedName: $base) {
+        compare(headRef: $head) {
```

Comment:
`compare` field is only supported in `ref`.

For now we cannot compare commits:

```graphql
query compareBaseHead($owner: String!, $name: String!, $base: String!, $head: String!, $size: Int!, $after: String) {
  rateLimit {
    cost
  }
  repository(owner: $owner, name: $name) {
    ref(qualifiedName: $base) {
      compare(headRef: $head) {
        commits(first: $size, after: $after) {
          totalCount
          pageInfo {
            hasNextPage
            endCursor
          }
          nodes {
            committedDate
            oid
            associatedPullRequests(first: 1, orderBy: {field: CREATED_AT, direction: ASC}) {
              nodes {
                number
                author {
                  login
                }
              }
            }
          }
        }
      }
    }
  }
}
```

```json
{
  "owner": "int128",
  "name": "list-associated-pull-requests-action",
  "base": "639b198eeec3ff282b00e53e936929b72c004817",
  "head": "0f9bbfade221f8b590163c664e27c3839fffd3c8",
  "size": 100
}
```